controllers/storagecluster/platform_detection.go

```diff
@@ -138,12 +128,14 @@ func (r *StorageClusterReconciler) DevicesDefaultToFastForThisPlatform() (bool,
 // PlatformsShouldAvoidObjectStore determines whether an object store should be created
 // for the platform.
 func (r *StorageClusterReconciler) PlatformsShouldAvoidObjectStore() (bool, error) {
-	// Call GetPlatform to get platform
-	platform, err := r.platform.GetPlatform(r.Client)
-	if err != nil {
-		return false, err
+	// call detectPlatform func to detect the platform of running cluster
+	if err := r.platform.detectPlatform(r.Client); err != nil {
+		r.Log.Error(err, "Failed to get platform")
```

Comment:
Do we still need to run this here if we're already calling it when setting up the reconciler?

Reply:
the issue I was having was I added detectPlatform() func in storagecluster_controller.go but when I deployed the operator in aws I was still seeing rgw pod. Which shouldn’t have come up.

Reply:
@jarrpa the above line can be added in GetPlatform() func and can be removed in controller.go file

Reply:
If I'm understanding that correctly, that would just revert back to the behavior we had before, which I am explicitly asking to remove. It seems to be that we should (theoretically) be able to configure the reconciler before we ever run Reconcile(). That's the core issue I want to resolve.